packages/portal/src/components/search/SearchQueryBuilderRule.vue

```diff
@@ -149,56 +145,111 @@
 
     data() {
       return {
-        field: this.value.field,
-        modifier: this.value.modifier,
-        term: this.value.term
+        aggregatedFieldNames: ['who', 'where', 'when', 'what'],
+        field: null,
+        fulltextFieldName: 'fulltext',
+        modifier: null,
+        ruleComponents: ['field', 'modifier', 'term'],
+        term: null,
+        validations: {
+          field: { state: null },
+          modifier: { state: null },
+          term: { state: null }
+        }
       };
     },
 
     computed: {
-      // If any field has a value, all are required. If none have a value, the
-      // rule will be ignored and none are required.
-      areAllRequired() {
-        return [this.value.term, this.value.field, this.value.modifier].some((value) => !!value);
+      aggregatedFieldOptions() {
+        return this.fieldOptions
+          .filter((field) => this.aggregatedFieldNames.includes(field.value));
+      },
+      dropdownSections() {
```

Comment:
We don't strictly need to have these as computed properties, this could all just be in the data for now. It will never change at the moment. 
Computed properties help split it up a bit though and I think this will help us in future should we need to have field specific modifiers.

Reply:
As it's dependent on data properties from the advancedSearch mixin, I don't think it can go into data on the component.